075-auto-rebalancing-cluster-scaling.md

```diff
@@ -0,0 +1,502 @@
+The way the user has to move topics' partitions in the above scenarios is via the Cruise Control integration by using the `KafkaRebalance` custom resource.
+
+Currently, there is no automatic rebalancing when the user scales the cluster up or down.
+It means that:
+
+* after scaling the cluster up, in order to rebalance the cluster and move some topics' partitions to the newly added brokers, the user has to manually create a `KafkaRebalance` custom resource using the `spec.mode: add-brokers` mode with the new brokers' IDs set on the `spec.brokers` field.
+* before scaling the cluster down, in order to rebalance the cluster and move topics' partitions off the brokers to be removed, the user has to manually create a `KafkaRebalance` custom resource using the `spec.mode: remove-brokers` mode with the IDs of the brokers to be removed set on the `spec.brokers` field.
+
+For more details, the proposal [035](https://github.com/strimzi/proposals/blob/main/035-rebalance-types-scaling-brokers.md) introduced the rebalancing modes mentioned above, which will be also referenced across this proposal.
+
+## Motivation
+
+The rebalancing operation during scaling up and down is a manual two-step process:
+
+* Scaling up: 
+   1. Scaling the cluster up by adding brokers
+   2. Running a rebalance
+* Scaling down: 
+   1. Running a rebalance
+   2. Scaling the cluster down by removing brokers
+
+In order to simplify the process, it would be great for the user to deal with just scaling the cluster and then having the operator taking care of running a rebalancing operation automatically when needed.
+
+## Proposal
+
+This proposal is about providing the user the possibility to automatically run a rebalancing when they scale the cluster up (by adding brokers) or scale the cluster down (by removing brokers) by changing the `spec.replicas` within a `Kafka` or `KafkaNodePool` custom resource.
+
+### Extending the `Kafka.spec.cruiseControl` field with auto-rebalancing related configuration
+
+The `Kafka` custom resource is extended by adding a new `spec.cruiseControl.autoRebalance` field.
+By default, without such field, no auto-rebalancing runs on scaling.
+
+The `spec.cruiseControl.autoRebalance` field allows to specify a list of rebalancing modes with the corresponding configuration to be used.
+For each entry in the list, the mode is defined by the `spec.cruiseControl.autoRebalance.mode` field (`add-brokers` or `remove-brokers`) and the corresponding rebalancing configuration is defined as a reference to a "template" `KafkaRebalance` custom resource (read later for more details), by using the `spec.cruiseControl.autoRebalance.template` field as a [LocalObjectReference](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/local-object-reference/).
+This field is optional and if not specified, the auto-rebalancing runs with the default Cruise Control configuration.
```

Comment:
do we need to say something about the results of running the default config?

Reply:
wdym by "the results of running the default config"? It's going to use what's mentioned here https://strimzi.io/docs/operators/latest/deploying#default-goals